docs/design/velero-krm-enhancement.md

```diff
@@ -0,0 +1,103 @@
+---
+title: velero-openshiftkrm-plugin design
+authors:
+  - "@ChaitanyaBhangale"
+reviewers:
+  - "@jwmatthews" 
+  - "@shubham-pampattiwar"
+  - "@dylan-murray"
+  - "@tiger-kaovilai"
+approvers:
+  - "@jwmatthews" 
+  - "@dylan-murray"
+  - "@shubham-pampattiwar"
+creation-date: 2022-06-21
+last-updated: 2022-06-21
+status: 
+  - "N/A"
+see-also:
+  - "N/A"  
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# Velero KRM Plugin Design
+
+### Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+### Open Questions 
+-  Can changes be made in exising codebase as opposed to refactoring the whole code? 
+-  Whether adding KRM (Kubernetes Resource Management) functions is actually useful or will it affect the performance of current plugin in a negative manner?
```

Comment:
I don't think that we are worried about performance.

When you say are actually useful, I would hope that you could define ways in which it would be useful in the enhancement, in the motivation section specifically.